### PR TITLE
Add anime detail page UI

### PR DESCRIPTION
**Pull request by Shiney-Kelly**
## Work Summary
- Add anime detail screen
- Add get single anime by id method in viewmodel
- Add Picasso image handling
- Add mappings between api model and data model

### CODE REVIEW

File: app/src/main/java/com/project/ianime/utils/image/ImageUtils.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    /**
+     * Get encoded image bitmap with Base-64
+     * @param [bitmap] - bitmap of current upload image
+     */
+    fun getEncodedSelectedImageBitmap(bitmap: Bitmap): String {
+        val outputStream = ByteArrayOutputStream()
+        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, outputStream)
+        val bytes = outputStream.toByteArray()
+        return Base64.encodeToString(bytes, Base64.DEFAULT)
```
Comment (Tony-Feng) [inline app/src/main/java/com/project/ianime/utils/image/ImageUtils.kt:25]: this func is not in use?

File: app/src/main/java/com/project/ianime/utils/image/ImageUtils.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                    Picasso.get()
+                        .load(imageUrl)
+                        .placeholder(R.drawable.ic_anime_placeholder)
+                        .into(imageView, object : Callback {
+                            override fun onSuccess() {
+                                Log.i("Picasso", "Image saved in Cache successfully")
+                            }
+
+                            override fun onError(e: java.lang.Exception?) {
+                                Log.i("Picasso", "Could not fetch image")
```
Comment (Tony-Feng) [inline app/src/main/java/com/project/ianime/utils/image/ImageUtils.kt]: can add img url in log

File: app/src/main/res/values-zh/strings.xml [hunk 3]
```diff
@@ -6,15 +6,32 @@
...
     <!-- Anime Detail Screen -->
     <string name="intro_header_title">剧情简介</string>
     <string name="anime_type_title">分类: %1$s</string>
     <string name="anime_country_title">国家: %1$s</string>
-    <string name="anime_year_title">发行时间: %1$s</string>
+    <string name="anime_release_year_title">发行时间: %1$s</string>
     <string name="anime_status_title">状态: %1$s</string>
     <string name="maxim_header_title">打动的话</string>
     <string name="episode_header_title">高光集数</string>
+    <string name="empty_description_message">我们需要迷迷提供更多关于这个动漫的信息。。。</string>
```
Comment (Tony-Feng) [inline app/src/main/res/values-zh/strings.xml:20]: `迷迷`?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -5,16 +5,33 @@
...
     <!-- Anime Detail Screen -->
     <string name="intro_header_title">About</string>
     <string name="anime_type_title">Type: %1$s</string>
     <string name="anime_country_title">Country: %1$s</string>
-    <string name="anime_year_title">Published Year: %1$s</string>
+    <string name="anime_release_year_title">Published Year: %1$s</string>
     <string name="anime_status_title">Status: %1$s</string>
     <string name="maxim_header_title">Maxim</string>
     <string name="episode_header_title">Favourite Episodes</string>
+    <string name="empty_description_message">We need you input some more information for this anime …</string>
```
Comment (Tony-Feng) [inline app/src/main/res/values/strings.xml]: remove space and replace to `.`: `We need you to provide more information about this anime...`

**Review by Tony-Feng — CHANGES_REQUESTED** [review]

**Comment by Tony-Feng** [comment]
actually no need to squash commits, since it doesn't have limit on that

File: app/src/main/java/com/project/ianime/utils/image/ImageUtils.kt (hunk 1 above)
Comment (Shiney-Kelly) [reply app/src/main/java/com/project/ianime/utils/image/ImageUtils.kt:25]: leave it as backup

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1,20 +1,37 @@
...
     <!-- Anime Detail Screen -->
     <string name="intro_header_title">About</string>
     <string name="anime_type_title">Type: %1$s</string>
     <string name="anime_country_title">Country: %1$s</string>
-    <string name="anime_year_title">Published Year: %1$s</string>
+    <string name="anime_release_year_title">Published Year: %1$s</string>
     <string name="anime_status_title">Status: %1$s</string>
     <string name="maxim_header_title">Maxim</string>
     <string name="episode_header_title">Favourite Episodes</string>
+    <string name="empty_description_message" tools:ignore="TypographyEllipsis">We need you input some more information for this anime...</string>
```
Comment (Tony-Feng) [inline app/src/main/res/values/strings.xml]: you can directly use this: `We need you to provide more information about this anime...`

**Review by Tony-Feng — APPROVED** [review]
LGTM!

**Pull request merged**